modules/rostests/apitests/advapi32/RegCreateKeyEx.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for RegCreateKeyExW.
+ * COPYRIGHT:   Copyright 2023 Doug Lyons <douglyons@douglyons.com>
+ */
+
+/*
+ * Based on code from the following:
+ * https://learn.microsoft.com/en-us/windows/win32/secauthz/creating-a-security-descriptor-for-a-new-object-in-c--
+ */
+
+#include <apitest.h>
+#include <stdio.h>
+
+#define WIN32_NO_STATUS
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+#include <windef.h>
+#include <aclapi.h>
+
+
+START_TEST(RegCreateKeyEx)
+{
+    HKEY hkey_main;
+    DWORD dwRes, dwDisposition;
+    PACL pACL = NULL;
+    PSECURITY_DESCRIPTOR pSD = NULL;
+    PSID pEveryoneSID = NULL, pAdminSID = NULL;
+    SID_IDENTIFIER_AUTHORITY SIDAuthWorld = {SECURITY_WORLD_SID_AUTHORITY};
+    SID_IDENTIFIER_AUTHORITY SIDAuthNT = {SECURITY_NT_AUTHORITY};
+    EXPLICIT_ACCESSW ea[2];
+    SECURITY_ATTRIBUTES sa;
+    LONG lRes;
+    BOOL bRes;
+    LONG ErrorCode = 0;
+    HKEY hkSub = NULL;
+
+    /* If any of the test keys already exist, delete them to ensure proper testing*/
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey", KEY_ALL_ACCESS, 0);
+    }
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey1", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
```

Comment:
Why do you need 3 keys, instead of 1 only?

Reply:
This allows us to do our tests without having to be concerned as much about any interaction between our tests. Thanks.

Reply:
Less code = less testing. I would like this to remain as it is unless a senior developer disagrees. Thanks.